### PR TITLE
PAT-1926 offender-risk-profiler convert to Kotlin

### PR DESCRIPTION
**Pull request by steverendell**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  testImplementation("com.github.tomakehurst:wiremock-standalone:2.27.2")
+  testImplementation("com.google.code.gson:gson:2.8.7")
+  testImplementation("com.nhaarman:mockito-kotlin-kt1.1:1.6.0")
+  testImplementation("org.awaitility:awaitility-kotlin:4.1.0")
+}
+
+tasks {
+  compileKotlin {
+    // kotlinOptions.jvmTarget = "16"
+  }
```
Comment (garethmdavies) [inline build.gradle.kts:87]: Can this be uncommented?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/riskprofiler/config/CacheConfig.kt [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  private val password: String? = null
+
+  @Value("\${spring.redis.ssl}")
+  private val ssl = false
+
+  @Value("\${spring.redis.client-name}")
+  private val clientName: String? = null
+
+  @Bean
+  fun jedisConnectionFactory(): JedisConnectionFactory {
```
Comment (garethmdavies) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/riskprofiler/config/CacheConfig.kt:36]: Should this be **redis** not jedis?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/riskprofiler/dao/OcgRepository.kt [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+          }
+        } catch (e: Exception) {
+          log.warn("Error in Line {} data [{}]", data.index, p)
+          data.linesError.incrementAndGet()
+        }
+      }
+    log.info(
+      "Lines total {}, processed {}, dups {}, invalid {}, errors {}", data.index.get(),
+      data.linesProcessed.get(), data.linesDup.get(), data.linesInvalid.get(), data.linesError.get()
+    )
```
Comment (lbennett-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/riskprofiler/dao/OcgRepository.kt:47]: kotlin interpolation is nice - log.info("Your username is ${user.username}") or without the brackets for simple 
strings

**Review by lbennett-moj — DISMISSED** [review]

File: src/main/kotlin/uk/gov/justice/digital/hmpps/riskprofiler/camel/CsvProcessorRoute.kt [hunk 4]
```diff
@@ -0,0 +1,54 @@
+package uk.gov.justice.digital.hmpps.riskprofiler.camel
+
+import org.apache.camel.builder.RouteBuilder
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.riskprofiler.services.DataService
+import uk.gov.justice.digital.hmpps.riskprofiler.services.FileService
+
+@Component
+// @DependsOn("dataRepositoryFactory")
```
[the remark this answers is not in the document]
Comment (steverendell) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/riskprofiler/camel/CsvProcessorRoute.kt:9]: can remove 👍

File: src/main/kotlin/uk/gov/justice/digital/hmpps/riskprofiler/dao/OcgRepository.kt (hunk 3 above)
Comment (steverendell) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/riskprofiler/dao/OcgRepository.kt:47]: good idea 👍

File: build.gradle.kts (hunk 1 above)
Comment (steverendell) [reply build.gradle.kts:87]: Turns out not, as testing uses a machine image which has an earlier version of java. There isnt an image available with java 16 yet :(

**Review by garethmdavies — APPROVED** [review]
LGTM

**Pull request merged**